### ISSUE TITLE
Claims retrying download, however when exact query is run via option 4, there are plenty of links

### ISSUE THREAD

**Issue by kent-chan-dev** [itsToggle/plex_debrid#638]
![Screenshot 2024-02-01 193837](https://github.com/itsToggle/plex_debrid/assets/51689757/075d8e2d-7cdf-42f9-b680-eb60b93891de)
When searching for mission impossible dead reckoning 2023, the script has found 31 releases, but apparently it is not able to pick a link. 
I am not sure why this is the case, since when I run the same query mission.impossible.dead.reckoning.2023, I am manually able to select one that fits the criteria, which is 5th from the list down. 

What could be the issue?
Also, how do I scroll the list of options when using the manual scraper?

**Comment by ghost** [comment]
Even I am facing the same issue recently. I thought there was some issue with my server even though I had not made any changes to my server. 
<img width="960" alt="Screenshot 2024-02-03 at 18 26 46" src="https://github.com/itsToggle/plex_debrid/assets/132819104/1ec11858-4cff-4681-ba30-f23039f5e91d">
Please see the attached screenshot above. I have recloned the repo and deleted the settings file just to see that others are also facing the same issue.

**Comment by TheIndieArmy** [comment]
This looks to be the same issue as 529: https://github.com/itsToggle/plex_debrid/issues/529

Hoping they fix it soon. It seems like a major issue that's affecting the main purpose of this app.

**Comment by ghost** [comment]
Yes but now only it has become persistent. I hope no expired link is being used in the script. I wonder if anyone else is facing this issue not reporting it yet.

**Comment by nobbylark** [comment]
Anyone manage to fix this issue? I'm facing the same.

**Comment by sirstudly** [comment]
Try it again with this branch:
```git clone -b issue_638 https://github.com/sirstudly/plex_debrid.git```

**Comment by Haroon01** [comment]
> Try it again with this branch: `git clone -b issue_638 https://github.com/sirstudly/plex_debrid.git`

it didnt work for me. im having the same problem. Using a residential IP, personal computer. Not a datacenter

![image](https://github.com/itsToggle/plex_debrid/assets/27559668/3027bcf4-ec4a-4838-a9d1-d274e2d40984)

**Comment by ghost** [comment]
Hi everyone, whoever struggling with this error please make sure that you are not running this script on a VPS as RealDebrid blocking so many IPs from VPS providers. I had to reset my server to figure out this issue. I am using VPN now and tunneling all my traffic through it. Its working fine for me now. The branch created by sirstudly was not working for me as well. Only after resetting the server and installing a VPN client solved the issue. If you can proxy the RealDebrid traffic through residential address will also resolve this error.

**Comment by nobbylark** [comment]
> Try it again with this branch: `git clone -b issue_638 https://github.com/sirstudly/plex_debrid.git`

This worked for me ! Cheers man.

**Comment by ghost** [comment]
I have installed it using the main branch itself but could you please explain if this is the issue that you are targeting in your project? 

https://www.reddit.com/r/RealDebrid/comments/13grcbj/realdebrid_now_blocking_servers/?rdt=51683

Recently RealDebrid started blocking Datacenter IPs and that is the reason why the script is not working unless it is redirected. @sirstudly could you please confirm that you are targeting this exact issue and hence your script can be used without any issue? I tried but it was giving me same error.

**Comment by sirstudly** [comment]
> I have installed it using the main branch itself but could you please explain if this is the issue that you are targeting in your project?
> 
> https://www.reddit.com/r/RealDebrid/comments/13grcbj/realdebrid_now_blocking_servers/?rdt=51683
> 
> Recently RealDebrid started blocking Datacenter IPs and that is the reason why the script is not working unless it is redirected. @sirstudly could you please confirm that you are targeting this exact issue and hence your script can be used without any issue? I tried but it was giving me same error.

No, the issue I was addressing was actually to do with [#529](https://github.com/itsToggle/plex_debrid/issues/529): punctuation marks in the release title. I've made an additional comment on the pull request to clarify this. Apologies for the confusion.

**Comment by skearney414** [comment]
This is mostly caused by the torrent not being cached on real-debrid. Try this image skearney414/plex_debrid. If the torrent is uncached you will see it in the logs

**Comment by jeffss00** [comment]
> This is mostly caused by the torrent not being cached on real-debrid. Try this image skearney414/plex_debrid. If the torrent is uncached you will see it in the logs 

@skearney414 Interesting, thanks! Will it download the torrent to real-debrid if it detects its not cached or will it just keep retrying until it shows up cached?

**Comment by TheIndieArmy** [comment]
> > This is mostly caused by the torrent not being cached on real-debrid. Try this image skearney414/plex_debrid. If the torrent is uncached you will see it in the logs
> 
> @skearney414 Interesting, thanks! Will it download the torrent to real-debrid if it detects its not cached or will it just keep retrying until it shows up cached?

I believe that depends on if you have cached set to required in your version settings or not.
